File: sample-videochat-webrtc/sample-videochat-webrtc/ViewControllers/Settings/SliderItemModel.m
//
//  SliderItemModel.m
//  sample-videochat-webrtc
//
//  Created by Andrey Ivanov on 30/09/15.
//  Copyright © 2015 QuickBlox Team. All rights reserved.
//

#import "SliderItemModel.h"
#import "SettingSliderCell.h"

@implementation SliderItemModel

- (instancetype)initWithTitle:(NSString *)title minValue:(NSUInteger)minValue maxValue:(NSUInteger)maxValue currentValue:(NSUInteger)currentValue {
    self = [super initWithTitle:title];
    if (self) {
        self.title = title;
        self.minValue = minValue;
        self.maxValue = maxValue;
        self.currentValue = currentValue;
    }
    return self;
}

- (instancetype)initWithMinValue:(NSUInteger)minValue maxValue:(NSUInteger)maxValue currentValue:(NSUInteger)currentValue {
    return [self initWithTitle:@"" minValue:minValue maxValue:maxValue currentValue:currentValue];
}

- (Class)viewClass {
    
    return [SettingSliderCell class];
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Settings/SliderItemModel.swift
//
//  SliderItemModel.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 22.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import Foundation

class SliderItemModel: BaseItemModel {
    var minLabel = ""
    var maxLabel = ""
    var maxValue: Int = 0
    var currentValue: UInt = 0
    var minValue: Int = 0
    var isDisabled = false
    
    override func viewClass() -> AnyClass {
        
        return SettingSliderCell.self
    }
}
